### ISSUE TITLE
[FORMATTING] Can't handle trino between date clause

### ISSUE THREAD

**Issue by rigogsilva** [sql-formatter-org/sql-formatter#507]
**Input data**

Which SQL and options did you provide as input?

```js
    format(query, {
      language: "trino",
      tabWidth: 2,
      logicalOperatorNewline: "before"
    })
```

```sql
SELECT * from test
  AND date_field BETWEEN DATE '2022-06-01' AND DATE '2022-08-09'
```

**Expected Output**

Any formatting.

**Usage**

- How are you calling / using the library?
```js
    format(query, {
      language: "trino",
      tabWidth: 2,
      logicalOperatorNewline: "before"
    })
```

- What SQL language(s) does this apply to?
trino

- Which SQL Formatter version are you using?
`"sql-formatter": "^11.0.2"`

error: 
```
Command 'trinols: Format' resulted in an error (Parse error at token: '2022-06-01' at line 2 column 31
Unexpected STRING token: {"type":"STRING","raw":"'2022-06-01'","text":"'2022-06-01'","start":49,"precedingWhitespace":" "}. Instead, I was expecting to see one of the following:

A AND token based on:
between_predicate → %BETWEEN _ expression _ ● %AND _ expression
asteriskless_free_form_sql$subexpression$1 →  ● 
```

**Comment by nene** [comment]
Thanks for reporting.

Unfortunately the date/time literal syntax is not correctly handled, plus the parsing of BETWEEN operator is somewhat crappy. I guess this syntax would have worked in some earlier version before the BETWEEN-expression parsing was rewritten, but then again these earlier versions don't have Trino support.

Unfortunately no quick fix for that. I'm working on a longer-term solution, but that will take some time.

**Comment by nene** [comment]
This should now be fixed in 12.0.2 release.
